aci/resource_aci_vmmvswitchpolicycont.go

```diff
@@ -0,0 +1,433 @@
+				Type: schema.TypeString,
+
+				Optional:    true,
+				Description: "Create relation to mcpIfPol",
+			},
+			"relation_vmm_rs_vswitch_override_cdp_if_pol": &schema.Schema{
+				Type: schema.TypeString,
+
+				Optional:    true,
+				Description: "Create relation to cdpIfPol",
+			},
+			"relation_vmm_rs_vswitch_override_lacp_pol": &schema.Schema{
+				Type: schema.TypeString,
+
+				Optional:    true,
+				Description: "Create relation to lacpLagPol",
+			},
+			"relation_vmm_rs_vswitch_override_mtu_pol": &schema.Schema{
+				Type: schema.TypeString,
+
+				Optional:    true,
+				Description: "Create relation to l2InstPol",
+			},
+		}),
+	}
+}
+
+func getRemoteVSwitchPolicyGroup(client *client.Client, dn string) (*models.VSwitchPolicyGroup, error) {
+	vmmVSwitchPolicyContCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+
+	vmmVSwitchPolicyCont := models.VSwitchPolicyGroupFromContainer(vmmVSwitchPolicyContCont)
+
+	if vmmVSwitchPolicyCont.DistinguishedName == "" {
+		return nil, fmt.Errorf("VMM vSwitch Policy %s not found", vmmVSwitchPolicyCont.DistinguishedName)
+	}
+
+	return vmmVSwitchPolicyCont, nil
+}
+
+func setVSwitchPolicyGroupAttributes(vmmVSwitchPolicyCont *models.VSwitchPolicyGroup, d *schema.ResourceData) *schema.ResourceData {
+	d.SetId(vmmVSwitchPolicyCont.DistinguishedName)
+	d.Set("description", vmmVSwitchPolicyCont.Description)
+	// d.Set("vmm_domain_dn", GetParentDn(vmmVSwitchPolicyCont.DistinguishedName))
+	vmmVSwitchPolicyContMap, _ := vmmVSwitchPolicyCont.ToMap()
+	// d.Set("name", vmmVSwitchPolicyContMap["name"])
+	d.Set("vmm_domain_dn", GetParentDn(vmmVSwitchPolicyCont.DistinguishedName, "/vswitchpolcont"))
+
+	d.Set("annotation", vmmVSwitchPolicyContMap["annotation"])
+	d.Set("name_alias", vmmVSwitchPolicyContMap["nameAlias"])
+	return d
+}
+
+func resourceAciVSwitchPolicyGroupImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+
+	aciClient := m.(*client.Client)
+
+	dn := d.Id()
+
+	vmmVSwitchPolicyCont, err := getRemoteVSwitchPolicyGroup(aciClient, dn)
+
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled := setVSwitchPolicyGroupAttributes(vmmVSwitchPolicyCont, d)
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciVSwitchPolicyGroupCreate(d *schema.ResourceData, m interface{}) error {
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+	VMMDomainDn := d.Get("vmm_domain_dn").(string)
+
+	vmmVSwitchPolicyContAttr := models.VSwitchPolicyGroupAttributes{}
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		vmmVSwitchPolicyContAttr.Annotation = Annotation.(string)
+	}
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		vmmVSwitchPolicyContAttr.NameAlias = NameAlias.(string)
+	}
+	vmmVSwitchPolicyCont := models.NewVSwitchPolicyGroup(fmt.Sprintf("vswitchpolcont"), VMMDomainDn, desc, vmmVSwitchPolicyContAttr)
+
+	err := aciClient.Save(vmmVSwitchPolicyCont)
+	if err != nil {
+		return err
+	}
+
```

Comment:
There should one pre check for all relations. That will check whether relation DN actually exist or not.